airbyte-workers/src/main/java/io/airbyte/workers/config/ContainerOrchestratorConfigBeanFactory.java

```diff
@@ -107,6 +110,10 @@ public ContainerOrchestratorConfig kubernetesContainerOrchestratorConfig(
       environmentVariables.put(DD_VERSION_ENV_VAR, System.getenv(DD_VERSION_ENV_VAR));
     }
 
+    if (System.getenv(DD_CONNECTOR_ENABLE_ENV_VAR) != null) {
+      environmentVariables.put(DD_CONNECTOR_JAVA_OPTS_ENV_VAR, System.getenv(containerConnectorJavaOpts));
```

Comment:
We also need a filter somewhere in `KubePodProcess` to ensure that this env var is only passed to the source connector image IF it's one that has the agent present.